### PR TITLE
74 Add component certification levels

### PR DESCRIPTION
**Pull request by sgnn7**
This PR adds:
- Certification badges w/ links to release notes' `Components` section
- Adds ability to use template helper functions
- Adds ability to use template partials/fragments
- Swaps from `html/template` to `text/template` package use (fixes non-ASCII escaping issues)
- Adds known certification values to `repositories.yml`
- Fixes problem with Conjur CLI version in `repositories.yml`

### CODE REVIEW

**Comment by sgnn7** [comment]
After this change RELEASE_NOTES.md looks like this:
-----
# Release Notes
All notable changes to this project will be documented in this file.

## [Unreleased] - 2020-03-05

## Table of Contents

- [Components](#components)
- [Upgrade Instructions](#upgrade-instructions)
- [Changes](#changes)

## Components

These are the components that combine to create this Conjur OSS Suite release and links
to their releases:

- **[cyberark/conjur v1.4.6](https://github.com/cyberark/conjur/releases/tag/v1.4.6)** (2020-01-21) [Certification Level](https://github.com/cyberark/conjur)
- **[cyberark/conjur-oss-helm-chart v1.3.7](https://github.com/cyberark/conjur-oss-helm-chart/releases/tag/v1.3.7)** (2019-01-31) [Certification Level](https://github.com/cyberark/conjur-oss-helm-chart)
- **[cyberark/secretless-broker v1.4.2](https://github.com/cyberark/secretless-broker/releases/tag/v1.4.2)** (2020-01-08) [Certification Level](https://github.com/cyberark/secretless-broker)
- **[cyberark/conjur-cli v6.2.1](https://github.com/cyberark/conjur-cli/releases/tag/v6.2.1)** (2019-05-22) [Certification Level](https://github.com/cyberark/conjur-cli)
- **[cyberark/conjur-api-go v0.6.0](https://github.com/cyberark/conjur-api-go/releases/tag/v0.6.0)** (2019-03-04) [Certification Level](https://github.com/cyberark/conjur-api-go)
- **[cyberark/conjur-api-java v2.0.0](https://github.com/cyberark/conjur-api-java/releases/tag/v2.0.0)** (2018-07-12) [Certification Level](https://github.com/cyberark/conjur-api-java)
- **[cyberark/conjur-api-python3 v0.0.5](https://github.com/cyberark/conjur-api-python3/releases/tag/v0.0.5)** (2019-12-06) [Certification Level](https://github.com/cyberark/conjur-api-python3)
- **[cyberark/conjur-api-ruby v4.10.2](https://github.com/cyberark/conjur-api-ruby/releases/tag/v4.10.2)** (2014-09-22) [Certification Level](https://github.com/cyberark/conjur-api-ruby)

## Upgrade Instructions

Upgrade instructions are available for the following components:

- [cyberark/conjur](https://docs.cyberark.com/Product-Doc/OnlineHelp/AAM-DAP/Latest/en/Content/Deployment/Upgrade/upgrade-intro.htm)

## Changes

The following are changes to the constituent components since the last Conjur
OSS Suite release:

### [cyberark/conjur v1.3.6](https://github.com/cyberark/conjur/releases/tag/v1.3.6) (2019-02-19)

#### Changed
- Reduced IAM authentication logging
- Refactored authentication strategies

#### Removed
- Removed OIDC APIs public access

### [cyberark/conjur v1.4.4](https://github.com/cyberark/conjur/releases/tag/v1.4.4) (2019-12-19)

#### Added
- Early validation of account existence during OIDC authentication
- Code coverage reporting and collection

#### Changed
- Bumped puma from 3.12.0 to 3.12.2
- Bumped rack from 1.6.11 to 1.6.12
- Bumped excon from 0.62.0 to 0.71.0

#### Fixed
- Fixed password rotation of blank password
- Fixed bug with multi-cert CA chains in Kubernetes service accounts
- Fixed build issues with creating namespaces with multiple values

#### Removed
- Removed follower env configuration

### [cyberark/conjur v1.4.6](https://github.com/cyberark/conjur/releases/tag/v1.4.6) (2020-01-21)

#### Changed
- K8s hosts' application identity is extracted from annotations or id. If it is
defined in annotations it will taken from there and if not, it will be taken
from the id.

### [cyberark/conjur-oss-helm-chart v1.3.7](https://github.com/cyberark/conjur-oss-helm-chart/releases/tag/v1.3.7) (2019-01-31)

#### Changed
- Server ciphers have been upgraded to TLS1.2 levels.

### [cyberark/secretless-broker v1.4.2](https://github.com/cyberark/secretless-broker/releases/tag/v1.4.2) (2020-01-08)

#### Added
- Updated CONTRIBUTING.md with instructions for using go-mssqldb submodule (#1044)
- Added gosec security scan to pipeline (#976)
- Added integration tests for MSSQL against additional MSSQL versions (#1017)
- Added gofmt to CodeClimate checks (#1055)
- Added support for MSSQL client parameter propagation (#1012)

#### Changed
- Bumped the conjur-authn-k8s-client version for the Conjur provider k8s
authenticator to v0.15.0 (#1060)
- Example plugin updated for clarity (#1061)
- Plugin SDK templates updated for clarity (#1054)
- Removed hardcoded PreloginResponse from MSSQL connector (#1014)
- Bumped Go version in Dockerfile to 1.13

#### Fixed
- Secretless doesn't exit when it can't start a configured connector (#1057)
- Secretless has insufficient logs when the config file has trouble loading (#1062)

### [cyberark/conjur-cli v6.2.1](https://github.com/cyberark/conjur-cli/releases/tag/v6.2.1) (2019-05-22)

#### Added
- Pin to xdg gem v2.2.3 due to a [crashing CLI](https://github.com/cyberark/conjur-cli/issues/243).

### [cyberark/conjur-api-go v0.6.0](https://github.com/cyberark/conjur-api-go/releases/tag/v0.6.0) (2019-03-04)

#### Added
- Converted to Golang 1.12
- Started using os.UserHomeDir() built-in instead of go-homedir module

### [cyberark/conjur-api-java v2.0.0](https://github.com/cyberark/conjur-api-java/releases/tag/v2.0.0) (2018-07-12)

#### Added
- License updated to Apache v2 - [PR #8](https://github.com/cyberark/conjur-api-java/pull/8)

#### Changed
- Authn tokens now use the new Conjur 5 format - [PR #21](https://github.com/cyberark/conjur-api-java/pull/21)
- Configuration change. When using environment variables, use CONJUR_AUTHN_LOGIN and CONJUR_AUTHN_API_KEY now instead of CONJUR_CREDENTIALS - [https://github.com/cyberark/conjur-api-java/commit/60344308fc48cb5380c626e612b91e1e720c03fb](https://github.com/cyberark/conjur-api-java/commit/60344308fc48cb5380c626e612b91e1e720c03fb)

### [cyberark/conjur-api-python3 v0.0.5](https://github.com/cyberark/conjur-api-python3/releases/tag/v0.0.5) (2019-12-06)

#### Added
- Added ability to delete policies [#23](https://github.com/cyberark/conjur-api-python3/issues/23)

### [cyberark/conjur-api-ruby v4.10.2](https://github.com/cyberark/conjur-api-ruby/releases/tag/v4.10.2) (2014-09-22)

#### Added
- Authn token is refetched before the expiration
- Support for configuration sticky option is discarded
- Resource#exists? refactored -- no overloading, code from exists.rb used
- Tests use Rspec v3 and reset configuration between test cases

File: pkg/template/template.go [hunk 1]
```diff
@@ -13,11 +15,13 @@ import (
...
-	ReleaseName string
-	ReleaseDate string
-	UpgradeURL  string
+	Repo               string
+	URL                string
+	ReleaseName        string
+	ReleaseDate        string
+	CertificationLevel string
+	Changelogs         []*changelog.VersionChangelog
+	UpgradeURL         string
```
Comment (BradleyBoutcher) [inline pkg/template/template.go]: Tiny nit: Organizing these could be nice 
```suggestion
	CertificationLevel  string
	Changelogs           []*changelog.VersionChangelog
	Repo                      string
	ReleaseName        string
	ReleaseDate          string
	UpgradeURL          string
	URL                        string
```

File: pkg/template/template.go [hunk 2]
```diff
@@ -28,12 +32,37 @@ type ReleaseSuite struct {
...
+type Engine struct {
+	BaseDir string
+}
+
+// Define helper methods for templates
+var funcMap = stdlibTemplate.FuncMap{
+	"toLower": strings.ToLower,
+}
+
+// New returns a new templating.Engine based on the specified root
```
Comment (BradleyBoutcher) [inline pkg/template/template.go]: love this

File: pkg/template/template_test.go [hunk 3]
```diff
@@ -54,7 +59,8 @@ func TestWriteChangelogDestinationOpenError(t *testing.T) {
 	outputFile := "doesnotexist/foo"
 	testObj := struct{}{}
 
-	err := WriteChangelog("testdata/test.tmpl", testObj, outputFile)
+	tmpl := New("testdata")
+	err := tmpl.WriteChangelog("test.tmpl", testObj, outputFile)
 	if !assert.Error(t, err) {
 		return
 	}
```
Comment (BradleyBoutcher) [inline pkg/template/template_test.go]: Nit: We could probably cut down on code reuse with an array of test cases that we iterate over

File: cmd/changelog-parser/testdata/repositories.yml [hunk 4]
```diff
@@ -13,14 +13,17 @@ section:
         version: v1.4.6
         after: v1.3.5
         upgrade_url: https://docs.cyberark.com/Product-Doc/OnlineHelp/AAM-DAP/Latest/en/Content/Deployment/Upgrade/upgrade-intro.htm
+        certification: "trusted"
```
Comment (izgeri) [inline cmd/changelog-parser/testdata/repositories.yml]: This should be certified

File: cmd/changelog-parser/testdata/repositories.yml [hunk 5]
```diff
@@ -13,14 +13,17 @@ section:
...
       - name: cyberark/conjur-oss-helm-chart
         url: https://github.com/cyberark/conjur-oss-helm-chart
         description: Helm chart for deploying Conjur OSS.
         version: v1.3.7
+        certification: "certified"
```
Comment (izgeri) [inline cmd/changelog-parser/testdata/repositories.yml]: This should be trusted

File: cmd/changelog-parser/testdata/repositories.yml (hunk 4 above)
Comment (sgnn7) [reply cmd/changelog-parser/testdata/repositories.yml]: This is just test data

File: cmd/changelog-parser/testdata/repositories.yml (hunk 5 above)
Comment (sgnn7) [reply cmd/changelog-parser/testdata/repositories.yml]: Ditto - also just test data for the integration test

File: pkg/template/template_test.go (hunk 3 above)
Comment (sgnn7) [reply pkg/template/template_test.go]: Good point - this might be something too look into in the next iteration.

**Comment by sgnn7** [comment]
@BradleyBoutcher I approved your suggested changed which invalidated your approval. Do you mind re-approving when you get a chance?

**Pull request merged**